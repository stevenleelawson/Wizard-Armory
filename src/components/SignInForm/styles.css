@import url('https://fonts.googleapis.com/css?family=Lora');

.sign-in {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 1.2rem;
}

.sign-in-input {
  width: 15vw;
  margin: 1rem;
  border: none;
  border-bottom: 1px solid #4C1502;
}

.sign-in-btn {
  width: 15.1vw;
  border: .1px solid #4C1502;
  font-family: 'Lora', serif;
  margin-top: 1.5rem;
}

.sign-in-btn:hover {
  background: #F3C969;
}
