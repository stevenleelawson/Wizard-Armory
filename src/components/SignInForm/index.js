import React, { Component } from 'react';
import { auth } from '../../firebase';
import './styles.css';

class SignInForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      errorMsg: false
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { email, password, errorMsg } = this.state;
    auth.doSignInWithEmailAndPassword(email, password).catch((error) => this.setState({errorMsg: 'Please create an account'}));
      this.setState({
        email: '',
        password: ''
      })
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    })
  }

  render() {
    const { email, password, errorMsg } = this.state;
    return (
      <div className='sign-in-container'>
        <h5>Sign In</h5>
        <form className='sign-in' onSubmit={this.handleSubmit}>
          <input
            type='email'
            name='email'
            placeholder='email'
            value={email}
            onChange={this.handleChange}
            className='sign-in-input'
          />
          <input
            type='password'
            name='password'
            placeholder='password'
            value={password}
            onChange={this.handleChange}
            className='sign-in-input'
          />
          <button className='sign-in-btn' type='submit'>ENTER</button>
        </form>
        {errorMsg && <p>{errorMsg}</p>}
      </div>
    )
  }
}

export default SignInForm;
