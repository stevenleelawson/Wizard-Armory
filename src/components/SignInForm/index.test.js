import React from 'react';
import { shallow } from 'enzyme';
import SignInForm from './index';

describe('SignInForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SignInForm />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update the state on handleChange', () => {
    //setup
    const mockEvent = {
      target: {
        value: 'taco@taco.com',
        name: 'email'
      }
    }
    const mockUser = {
      email: 'taco@taco.com',
      password: '',
      errorMsg: false
    }
    //execution
    wrapper.instance().handleChange(mockEvent)

    //assertion
    expect(wrapper.state()).toEqual(mockUser)
  });
});
