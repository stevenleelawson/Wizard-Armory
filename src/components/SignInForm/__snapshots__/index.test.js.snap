// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignInForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Sign In
</h5>,
        <form
          className="sign-in"
          onSubmit={[Function]}
>
          <input
                    className="sign-in-input"
                    name="email"
                    onChange={[Function]}
                    placeholder="email"
                    type="email"
                    value=""
          />
          <input
                    className="sign-in-input"
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
          />
          <button
                    className="sign-in-btn"
                    type="submit"
          >
                    ENTER
          </button>
</form>,
        false,
      ],
      "className": "sign-in-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign In",
        },
        "ref": null,
        "rendered": "Sign In",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="sign-in-input"
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
/>,
            <input
              className="sign-in-input"
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
/>,
            <button
              className="sign-in-btn"
              type="submit"
>
              ENTER
</button>,
          ],
          "className": "sign-in",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sign-in-input",
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sign-in-input",
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ENTER",
              "className": "sign-in-btn",
              "type": "submit",
            },
            "ref": null,
            "rendered": "ENTER",
            "type": "button",
          },
        ],
        "type": "form",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Sign In
</h5>,
          <form
            className="sign-in"
            onSubmit={[Function]}
>
            <input
                        className="sign-in-input"
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="email"
                        value=""
            />
            <input
                        className="sign-in-input"
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                        value=""
            />
            <button
                        className="sign-in-btn"
                        type="submit"
            >
                        ENTER
            </button>
</form>,
          false,
        ],
        "className": "sign-in-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="sign-in-input"
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="email"
                value=""
/>,
              <input
                className="sign-in-input"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
/>,
              <button
                className="sign-in-btn"
                type="submit"
>
                ENTER
</button>,
            ],
            "className": "sign-in",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "sign-in-input",
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "sign-in-input",
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ENTER",
                "className": "sign-in-btn",
                "type": "submit",
              },
              "ref": null,
              "rendered": "ENTER",
              "type": "button",
            },
          ],
          "type": "form",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
